napari/utils/colormaps/colormap_utils.py

```diff
@@ -745,14 +765,64 @@ def ensure_colormap(colormap: ValidColormapArg) -> Colormap:
     return AVAILABLE_COLORMAPS[name]
 
 
-def _colormap_from_colors(colors: ColorType) -> Optional[Colormap]:
+def _hexcolor_to_name(hexcolor):
+    """Returns the name of a colormap according to the input hex value.
+
+    In case the color is recognised (solid color in rgbcmyw),
+    The existing colormap name associated with that color is returned.
+    Otherwise, a custom name is produced (#RgbA -> custom-rrggbbaa).
+
+    Parameters
+    ----------
+    hexcolor : str
+        A hex color value (#rgb, #rgba, #rrggbb, #rrggbbaa)
+
+    Returns
+    -------
+    Name of colormap
+    """
+
+    # A dictionary of known colormaps with lowercase rrggbbaa values as keys
+    rgba_colormaps = {
+        '#ffffffff': 'gray',
+        '#ff0000ff': 'red',
+        '#00ff00ff': 'green',
+        '#0000ffff': 'blue',
+        '#00ffffff': 'cyan',
+        '#ff00ffff': 'magenta',
+        '#ffff00ff': 'yellow',
+    }
+
+    # Here we convert the input hexcolor into a rrggbbaa hexcolor
+    hexcolor_rgba = hexcolor
+    # hex value is '#rgb' or '#rgba', need to double repeat each character
+    if len(hexcolor) == 4 or len(hexcolor) == 5:
+        hexcolor_rgba = '#' + ''.join([c * 2 for c in hexcolor[1:]])
+
+    if len(hexcolor_rgba) == 7:
+        hexcolor_rgba += 'ff'
+
+    name = rgba_colormaps.get(hexcolor_rgba.lower())
+
+    # The hex value isn't associated with a known colormap, so make a custom name for it
+    if name is None:
+        name = 'custom-' + hexcolor[1:].lower()
+
+    return name
```

Comment:
```suggestion
    return rgba_colormaps.get(hexcolor_rgba.lower(), 'custom-' + hexcolor[1:].lower())
```

This part could be simplified

Reply:
Thank you, this is much nicer!